code/modules/food_and_drinks/restaurant/_venue.dm

```diff
@@ -67,12 +67,41 @@
 	current_visitors += new_customer
 
 /datum/venue/proc/order_food(mob/living/simple_animal/robot_customer/customer_pawn, datum/customer_data/customer_data)
-	return
+	var/order = pick_weight(customer_data.orderable_objects[venue_type])
+	var/image/food_image
+	var/food_line
+
+	if(ispath(order, /datum/custom_order)) // generate the special order
+		var/datum/custom_order/custom_order = new order(src)
+		food_image = custom_order.get_order_appearance(src)
+		food_line = custom_order.get_order_line(src)
+		. = custom_order.dispense_order()
```

Comment:
We're supposed to be avoiding using `.` like this now. See https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-